### PR TITLE
BugFix: Introducing Introducing Delim Joins and Delim_Get(s) should respect positionally

### PR DESCRIPTION
**Pull request by Tmonster**
fixes https://github.com/duckdblabs/duckdb-internal/issues/1807

In queries where a correlated subquery requires a delim get, we want to preserve positionally of columns. Operators can get moved around and when operators get moved around the positionally of columns can change. This is especially bad when you have set operations, where positionally is important. 

A delim_get is introduced with a cross product. The children of a cross product can be flipped around by the join order optimizer, affecting the positionally of the columns. With this fix, if a projecition wants to introduce a delim get, the delim get cross product is produced under it.

### CODE REVIEW

File: test/optimizer/case_simplification.test [hunk 1]
```diff
@@ -2,6 +2,31 @@
...
+          )
+          UNION BY NAME
+          (
+              SELECT inner2."FLOAT" from all_types inner2
+          )
+      ) IS DISTINCT FROM outer_alltypes."struct"
+  )
+FROM all_types outer_alltypes GROUP BY ALL;
+
+mode skip
```
Comment (lnkuiper) [inline test/optimizer/case_simplification.test:28]: Is the skip here intentional?

File: src/planner/subquery/flatten_dependent_join.cpp [hunk 2]
```diff
@@ -119,6 +119,8 @@ unique_ptr<LogicalOperator> FlattenDependentJoins::PushDownDependentJoinInternal
                                                                                  idx_t lateral_depth) {
 	// first check if the logical operator has correlated expressions
 	auto entry = has_correlated_expressions.find(*plan);
+	bool exit_projection = false;
+	unique_ptr<LogicalDelimGet> delim_scan = nullptr;
```
Comment (lnkuiper) [inline src/planner/subquery/flatten_dependent_join.cpp:123]: nitpick: `unique_ptr` is initialized to `nullptr` by default

**Review by lnkuiper — CHANGES_REQUESTED** [review]
Looks good! Just two minor comments:

**Review by lnkuiper — APPROVED** [review]
LGTM now, thanks!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**